src/core/contract_address/starknet_contract_address.rs

```diff
@@ -0,0 +1,371 @@
+        let external_functions_flatted = contract_class
+            .external_functions
+            .iter()
+            .flat_map::<Vec<MaybeRelocatable>, _>(|ext_func| ext_func.into())
+            .collect::<Vec<MaybeRelocatable>>();
+
+        let l1_handlers_flatted = contract_class
+            .l1_handlers
+            .iter()
+            .flat_map::<Vec<MaybeRelocatable>, _>(|l1_handler| l1_handler.into())
+            .collect::<Vec<MaybeRelocatable>>();
+
+        let constructors_flatted = contract_class
+            .constructors
+            .iter()
+            .flat_map::<Vec<MaybeRelocatable>, _>(|constructor| constructor.into())
+            .collect::<Vec<MaybeRelocatable>>();
+
+        let result = vec![
+            CairoArg::Single(contract_class.api_version),
+            CairoArg::Single(contract_class.n_external_functions),
+            CairoArg::Array(external_functions_flatted),
+            CairoArg::Single(contract_class.n_l1_handlers),
+            CairoArg::Array(l1_handlers_flatted),
+            CairoArg::Single(contract_class.n_constructors),
+            CairoArg::Array(constructors_flatted),
+            CairoArg::Single(contract_class.n_builtins),
+            CairoArg::Array(contract_class.builtin_list),
+            CairoArg::Single(contract_class.hinted_class_hash),
+            CairoArg::Single(contract_class.bytecode_length),
+            CairoArg::Array(contract_class.bytecode_ptr),
+        ];
+        CairoArg::Composed(result)
+    }
+}
+
+use std::{collections::HashMap, hash::Hash, path::Path};
+
+pub(crate) fn compute_class_hash(
+    contract_class: &ContractClass,
+) -> Result<Felt, ContractAddressError> {
+    // Since we are not using a cache, this function replace compute_class_hash_inner.
+    let program = load_program()?;
+    let contract_class_struct =
+        &get_contract_class_struct(&program.identifiers, contract_class)?.into();
+    let mut vm = VirtualMachine::new(false);
+    let mut runner = CairoRunner::new(&program, "all", false)
+        .map_err(|err| ContractAddressError::CairoRunner(err.to_string()))?;
+    runner.initialize_function_runner(&mut vm);
+
+    let mut hint_processor = BuiltinHintProcessor::new_empty();
+
+    // 188 is the entrypoint since is the __main__.class_hash function in our compiled program identifier.
+    // TODO: Looks like we can get this value from the identifier, but the value is a Felt.
+    // We need to cast that into a usize.
+    // let entrypoint = program.identifiers.get("__main__.class_hash").unwrap().pc.unwrap();
+    let hash_base: MaybeRelocatable = runner.add_additional_hash_builtin(&mut vm).into();
+
+    runner
+        .run_from_entrypoint(
+            188,
+            &vec![&hash_base.into(), contract_class_struct],
+            true,
+            &mut vm,
+            &mut hint_processor,
+        )
+        .map_err(|err| ContractAddressError::CairoRunner(err.to_string()))?;
```

Comment:
same suggestion as before, turn the CairoRunner exception in a subset of ContractAddressError, for simplicity.